tmo_monitor/gateway/base.py

```diff
@@ -2,15 +2,35 @@
 import re
 import subprocess
 import time
+import sys
 
 class ControllerBase:
   # functions that don't touch the API
-  def ping(self, ping_host, ping_count, ping_interval, interface = None):
+  def ping(self, ping_host, ping_count, ping_interval, interface = None, ping_6 = False):
     is_win = platform.system() == 'Windows'
-    ping_cmd = ['ping']
+    is_osx = platform.system() == 'Darwin'
+
+    ping_cmd = []
+    extra_flags = []
+
+    # Handle IPv6 support - use ping6 binary or ping -6 flag
+    ping_bin = 'ping'
+    if ping_6:
+      if is_osx:
+        ping_bin = 'ping6'
+      else:
```

Comment:
@highvolt-dev - this test does not work all the time on Linux. For example, on OpenWrt, there are 2 distinct binaries: `ping` for IPv4, and `ping6` for IPv6. `ping -6` is not a valid option:

```
# opkg list iputils-ping\*
iputils-ping - 20101006-1 - Program ping from iputils.
 Sends ICMP ECHO_REQUEST to network hosts (IPv4).
iputils-ping6 - 20101006-1 - Program ping6 from iputils.
 Sends ICMP ECHO_REQUEST to network hosts (IPv6).
# opkg files iputils-ping6
Package iputils-ping6 (20101006-1) is installed on root and has the following files:
/usr/bin/ping6
```

A better test may be to test for a `ping6` binary and fall back to `ping -6`.